llvm/include/llvm/Support/float128.h

```diff
@@ -11,7 +11,14 @@
 
 namespace llvm {
 
-#if defined(__clang__) && defined(__FLOAT128__) &&                             \
+#if defined(__aarch64__)
```

Comment:
I would expect we might need to be careful of systems that have long-double==double, such as windows and darwin targets.

Reply:
I've added a check for the mantissa size now which should help with this.